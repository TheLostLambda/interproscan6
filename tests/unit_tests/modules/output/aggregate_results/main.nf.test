nextflow_process {

    name "Test Process AGGREGATE_RESULTS"
    script "interproscan/modules/output/aggregate_results/main.nf"
    process "AGGREGATE_RESULTS"

    stage {
        copy "interproscan/scripts/output/aggregate_results.py"
        copy "tests/unit_tests/test_inputs/aggregate_results/cdd_matches.json"
        copy "tests/unit_tests/test_inputs/aggregate_results/hamap_matches.json"
    }

    test("Should pass") {

        when {
            params {
            }
            process {
                """
                input[0] = ["../../../meta/tests/unit_tests/test_inputs/aggregate_results/cdd_matches.json", "../../../meta/tests/unit_tests/test_inputs/aggregate_results/hamap_matches.json"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
