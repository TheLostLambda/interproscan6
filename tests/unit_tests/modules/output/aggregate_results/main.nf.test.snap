{
    "Should pass": {
        "content": [
            {
                "0": [
                    "results_aggregated.json:md5,fb58604aca6dba51bf4acbaa19edb238"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-02T20:40:57.449253"
    },
    "Should run successfully": {
        "content": [
            {
                "0": [
                    "results_aggregated.json:md5,fb58604aca6dba51bf4acbaa19edb238"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-11T13:50:20.084969"
    }
}